internal/pkg/api/handleCheckin.go

```diff
@@ -376,6 +376,10 @@ func convertActions(agentID string, actions []model.Action) ([]ActionResp, strin
 
 	respList := make([]ActionResp, 0, sz)
 	for _, action := range actions {
+		if action.Type == "POLICY_CHANGE" {
```

Comment:
Nit. This is adding an additional responsibility to `convertActions`, it doesn't only convert anymore, but it also filters out certain actions. This may lead to issues in the future if we reuse this method to purely convert actions.

Consider adding something like a `filterActions([]model.Action) []model.Action` method to filter out the actions that are not relevant here, and explicitly filter out when needed.